neurom/apps/tests/test_morph_stats.py

```diff
@@ -130,17 +130,18 @@ def test_eval_stats_applies_numpy_function():
 
 
 def test_extract_stats_single_neuron():
-    nrn = nm.load_neuron(Path(SWC_PATH, 'Neuron.swc'))
-    res = ms.extract_stats(nrn, REF_CONFIG)
-    assert_equal(set(res.keys()), set(REF_OUT.keys()))
-    # Note: soma radius is calculated from the sphere that gives the area
-    # of the cylinders described in Neuron.swc
-    assert_almost_equal(res['mean_soma_radius'], REF_OUT['mean_soma_radius'])
+    path = SWC_PATH / 'Neuron.swc'
+    for nrn in (nm.load_neuron(path), path, str(path)):
+        res = ms.extract_stats(nrn, REF_CONFIG)
+        assert_equal(set(res.keys()), set(REF_OUT.keys()))
+        # Note: soma radius is calculated from the sphere that gives the area
+        # of the cylinders described in Neuron.swc
+        assert_almost_equal(res['mean_soma_radius'], REF_OUT['mean_soma_radius'])
 
-    for k in ('all', 'axon', 'basal_dendrite', 'apical_dendrite'):
-        assert_equal(set(res[k].keys()), set(REF_OUT[k].keys()))
-        for kk in res[k].keys():
-            assert_almost_equal(res[k][kk], REF_OUT[k][kk], places=3)
+        for k in ('all', 'axon', 'basal_dendrite', 'apical_dendrite'):
+            assert_equal(set(res[k].keys()), set(REF_OUT[k].keys()))
+            for kk in res[k].keys():
+                assert_almost_equal(res[k][kk], REF_OUT[k][kk], places=3)
 
```

Comment:
you don t test : 
```
(isinstance(neurons, list) and all(isinstance(nrn, (str, Path)) for nrn in neurons))
```
even if the coverage is ok.

Reply:
(or maybe I missed it)

Reply:
line 134 of the test ;)

Reply:
Actually my initial though was : 
I would expect extract_stats to raise if `neurons` does not have the correct format. 
and you don t test a failing list

Reply:
I have removed this change. I think they are no longer relevant.